tower/src/balance/p2c/layer.rs

```diff
@@ -1,25 +1,35 @@
-use super::BalanceMake;
+use super::MakeBalance;
 use rand::{rngs::SmallRng, Rng, SeedableRng};
 use std::{fmt, marker::PhantomData};
 use tower_layer::Layer;
 
-/// Efficiently distributes requests across an arbitrary number of services
+/// Construct load balancers ([`Balance`]) over dynamic service sets ([`Discover`]) produced by the
+/// "inner" service in response to requests coming from the "outer" service.
+///
+/// This construction may seem a little odd at first glance. This is not a layer that takes
+/// requests and produces responses in the traditional sense. Instead, it is more like
+/// [`MakeService`](tower::make_service::MakeService) in that it takes service _descriptors_ and
+/// produces _services_. Since [`Balance`] spreads requests across a _set_ of services, the inner
+/// service should produce a [`Discover`], not just a single [`Service`], given a service
+/// descriptor.
```

Comment:
in Linkerd we tend to refer to "service descriptors" (and, in general, all `Request` types for a `MakeService`) as "targets". Might be good to standardize terminology for this?

this is also what the definition of `MakeService` calls them:
https://github.com/tower-rs/tower/blob/6a25d322b5160c1033e8741ff5c12b175a83bc21/tower/src/make/make_service.rs#L15

Reply:
Yeah, I went with "service descriptor" here because I feel as though "target" is a hopefully non-descriptive name. It doesn't tell me anything about what that parameter _does_. It's true that it _might_ be a target (like an IP address), but I don't think that is _generally_ true. I'd actually be in favor of renaming target everywhere, but that's a longer discussion :sweat_smile: 

In the meantime, how would you feel about putting ("the `Target`") in parentheses?